### PR TITLE
Add gr-symbolmapping recipe

### PR DESCRIPTION
**Pull request by jdemel**
GNU Radio and Python module for fast standardized symbol (de)mappers.

In digital communication, we rely on symbol mapping such as QPSK, 16QAM, etc. There exists a set of common definitions for these mappings in across a wide variety of standards (WiFi, LTE, 5G NR, ...). Unfortunately, GNU Radio does not implement this standard set. Further, we want fast and efficient demappers that employ efficient approximations for these common mappings.

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by jdemel** [comment]
@conda-forge/help-python-c This should be ready for review. I hope this is the correct team. Or should I get in touch with the cpp team?

I added a skip for windows builds for now. The Azure pipeline would just break in every corner before the package itself was even started to be compiled. Maybe, this can be changed at a later date.

File: recipes/gnuradio-symbolmapping/build.sh [hunk 1]
```diff
@@ -0,0 +1,17 @@
+#!/usr/bin/env bash
+
+mkdir build
+cd build
+
+cmake_config_args=(
+    -DCMAKE_BUILD_TYPE=Release
+    -DCMAKE_PREFIX_PATH=$PREFIX
+    -DCMAKE_INSTALL_PREFIX=$PREFIX
```
Comment (xhochy) [inline recipes/gnuradio-symbolmapping/build.sh:9]: These are part of CMAKE_ARGS and should not be duplicated.

File: recipes/gnuradio-symbolmapping/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,65 @@
+{% set oot_name = "symbolmapping" %}
+{% set name = "gnuradio-" + oot_name %}
```
Comment (xhochy) [inline recipes/gnuradio-symbolmapping/meta.yaml:2]: Please avoid introducing more Jinja variables than necessary
```suggestion
{% set name = "gnuradio-symbolmapping" %}
```

File: recipes/gnuradio-symbolmapping/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    - git
+    - ninja
+    - pkg-config  # [not win]
+    # cross-compilation requirements
+    - python                              # [build_platform != target_platform]
+    - cross-python_{{ target_platform }}  # [build_platform != target_platform]
+    - numpy                               # [build_platform != target_platform]
+    - pybind11                            # [build_platform != target_platform]
+    # Add extra build tool dependencies here
+
```
Comment (xhochy) [inline recipes/gnuradio-symbolmapping/meta.yaml:31]:
```suggestion
```

File: recipes/gnuradio-symbolmapping/meta.yaml [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+  host:
+    - python
+    - numpy >=1.22
+    - fmt
+    - spdlog
+    - pybind11
+    - gmp  # [not win]
+    - fftw
+    - volk >=2.5
+    - gmp  # [linux]
```
Comment (xhochy) [inline recipes/gnuradio-symbolmapping/meta.yaml:41]:
```suggestion
    - gmp  # [unix]
    - fftw
    - volk >=2.5
```

File: recipes/gnuradio-symbolmapping/meta.yaml [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+    # the following two entries are for generating builds against specific GR versions
+    - gnuradio-core  # [not gnuradio_extra_pin]
+    - gnuradio-core {{ gnuradio_extra_pin }}.*  # [gnuradio_extra_pin]
```
Comment (xhochy) [inline recipes/gnuradio-symbolmapping/meta.yaml:44]: Don't use the `[]` when you actually are not using the functionality.
```suggestion
    - gnuradio-core  # not gnuradio_extra_pin
    - gnuradio-core {{ gnuradio_extra_pin }}.*  # gnuradio_extra_pin
```

**Review by xhochy — CHANGES_REQUESTED** [review]

**Comment by jdemel** [comment]
@xhochy Thanks for your suggestions. I integrated them. Are these suggestions part of a guideline?

File: recipes/gnuradio-symbolmapping/meta.yaml (hunk 5 above)
Comment (jdemel) [reply recipes/gnuradio-symbolmapping/meta.yaml:41]: I suspect either this change, or the `- gmp  # [unix]` change broke the build. At least it seems to work again after i reverted those changes.

Comment (jdemel) [reply recipes/gnuradio-symbolmapping/meta.yaml:41]: I've tried all kinds of different combinations. The original version is the only one that works. Also, the original version basically expresses an XOR between the two. This should be the correct behavior.

Comment (carterbox) [reply recipes/gnuradio-symbolmapping/meta.yaml:41]: Where does the `gnuradio_extra_pin` variable get defined? You haven't included a 'conda_build_config.yaml', so `gnuradio_extra_pin` should have no value.

`gnuradio-core` is [pinned globally on conda-forge](https://github.com/conda-forge/conda-forge-pinning-feedstock/blob/5bac11c21f2a14b9100b9127848310df7721b25f/recipe/conda_build_config.yaml#L347) to the minor version, so ideally we want you to not pin it again.

**Review by carterbox — CHANGES_REQUESTED** [review]

File: recipes/gnuradio-symbolmapping/meta.yaml (hunk 5 above)
Comment (jdemel) [reply recipes/gnuradio-symbolmapping/meta.yaml:41]: I pulled these lines from the GNU Radio OOT template.
If I get this correctly, the way forward would be to just have one line
```yaml
- gnuradio-core
```
without anything extra. Would that be correct?

**Pull request closed without merging**